command/certificate/p12.go

```diff
@@ -96,26 +112,26 @@ func p12Action(ctx *cli.Context) error {
 		x509CAs = append(x509CAs, x509Bundle...)
 	}
 
-	var password string
 	var err error
-
-	if passwordFile := ctx.String("password-file"); passwordFile != "" {
-		password, err = utils.ReadStringPasswordFromFile(passwordFile)
-		if err != nil {
-			return err
+	var password string
+	if !ctx.Bool("no-password") {
+		if passwordFile := ctx.String("password-file"); passwordFile != "" {
+			password, err = utils.ReadStringPasswordFromFile(passwordFile)
+			if err != nil {
+				return err
+			}
 		}
-	}
 
-	if password == "" {
-		pass, err := ui.PromptPassword("Please enter a password to encrypt the .p12 file")
-		if err != nil {
-			return errors.Wrap(err, "error reading password")
+		if password == "" {
```

Comment:
Shouldn't we just fail here or accept that the user wanted to use an empty password. If the password file you selected was empty, it seems weird to then ask the user for another password.

Reply:
I left the functionality that was already there. But I think it's better to use `--no-password --insecure` if you really want an empty string as the password

Reply:
Then maybe we should just fail if the password file is empty? 

We don't normally check that value. We just use it (I think).

Reply:
That also match how other commands act `step crypto jwk create`